### DISCUSSION TITLE
The Eheim Thermocontrol+ e 300 is working as expected

### DISCUSSION THREAD

**Discussion by M1cN** [davidm-glitch/home-assistant-eheim-digital#2] [Show and tell]
Thank you for the effort you put into this.
I have tested the python script with the Thermocontrol+ e 300 and this is working perfect.

![afbeelding](https://github.com/davidm-glitch/home-assistant-eheim-digital/assets/7816672/d7f6dc73-deaa-4d9a-9c7d-9a2a96b2d619)

![afbeelding](https://github.com/davidm-glitch/home-assistant-eheim-digital/assets/7816672/747d9f26-ca98-46fa-8941-ac1ddb6e67dd)

**Comment by davidm-glitch** [comment]
Thank you for testing, happy it works. 

Does the script run reliable? Be aware that the eheim_digital/status shows the mqtt connection, the status inside heater should show the websocket connection of the device but that's not finished yet. ;)

**Comment by M1cN** [comment]
I have it running for a day now. So i will update after sometime. The
script seems to run stable so far.

The biggest challenge is the quality of the connectivity of the hardware
itself.



Op za 1 jul. 2023 14:14 schreef David ***@***.***>:

**Comment by davidm-glitch** [comment]
@webnar can you tell me how you are running the script? Inside HA?

**Comment by M1cN** [comment]
I have created a service that starts the script outside of docker (HA runs
in a container).

Until now it really runs very stable. The heater itself has very poor wifi
signaal so bad that most of the time i cannot access the web interface. But
even when i cannot access the interface the script still collects the
required data.


Op zo 2 jul. 2023 14:04 schreef David ***@***.***>:

**Comment by davidm-glitch** [comment]
Ah okay, I am currently trying to get this running from inside HA with pyscript or AppDaemon but without success. 

I will update the file later, I got the availability check of the websocket connection ready and working now :)

**Reply by M1cN** [reply]
The updated file is not working for me. It doesn't collect any data from the heater.

**Reply by davidm-glitch** [reply]
Did you see that the config is now included in the script? Maybe you have to fix your configuration.

**Comment by M1cN** [comment]
I did noticed that the config was included in the script.

Op ma 3 jul. 2023 21:52 schreef David ***@***.***>:

**Comment by davidm-glitch** [comment]
Do you have an error message? I tried the script again and it works fine for me.

**Comment by M1cN** [comment]
I dont get any error messages. When i check the logging it shows that the
heater is not coming online. Maybe it has something to do with the terrible
bad WiFi connection webinterface is also not loading. But the first version
of the script works perfectly.

Op di 4 jul. 2023 18:31 schreef David ***@***.***>:

**Comment by davidm-glitch** [comment]
Did you check the maximum water level on the heater?

**Comment by M1cN** [comment]
Yeah its touching the red button.

Op di 4 jul. 2023 19:27 schreef David ***@***.***>:

**Comment by davidm-glitch** [comment]
thats to far in the water, max should be the marked water line, maybe thats the issue?

**Comment by M1cN** [comment]
According to the manual the red button needs to touch the water surface.
Connectivity is better since i have it like this.

Op di 4 jul. 2023 19:31 schreef David ***@***.***>:

**Comment by davidm-glitch** [comment]
![image](https://github.com/davidm-glitch/home-assistant-eheim-digital/assets/59320226/57720d91-4dc2-45dd-8441-1cf524a99515)


try it like this ;)

**Comment by davidm-glitch** [comment]
@webnar any updates? Did it help?
